### DISCUSSION TITLE
How can I create a primitive with specific material using the MeshcatVisualizer

### DISCUSSION THREAD

**Discussion by lwygzh** [stack-of-tasks/pinocchio#2079] [Q&A]
Hi,

I am trying to visualize my robot trajectory using the `MeshcatVisualizer` provided by pinocchio. 

This is what I am currently doing

```python
import hppfcl
import numpy as np
import pinocchio

robot = pinocchio.RobotWrapper.BuildFromURDF(robot_urdf, package_dir)

box = hppfcl.Box(1, 1, 1)
placement = pinocchio.SE3(np.eye(3), np.array([0.8, 0.7, 0.5]))
geom_obj = pinocchio.GeometryObject("box", 0, 0, box, placement)
geom_obj.meshColor = np.array([1.0, 0.0, 0.0, 1.0])
robot.visual_model.addGeometryObject(geom_obj)

viz = pinocchio.MeshcatVisualizer(robot.model, robot.collision_model, robot.visual_model)
```

However, the box object does not consider any reflection. I was wondering if there is a way to set the material of the box to something like

```python
meshcat.geometry.MeshPhongMaterial(color=0xff0000, specular=0x111111, shininess=30)
```

Additionally, I noticed that `pinocchio.GeometryObject` is now deprecated. Is there a replacement?

Any help would be greatly appreciated!

**Comment by jorisv** [comment]
Hello @lwygzh,

Where did you see that `pinocchio.GeometryObject` is deprecated ?

It's not possible right now to attach custom material to an object.

I see two options :

- Add a method to `BaseVisualizer` to add custom `GeometryObject` with a Material as optional parameters
- Add a material attribute to `GeometryObject`

**Reply by lwygzh** [reply]
@jorisv Thanks for the response!

When I run

```
Python 3.8.15 | packaged by conda-forge | (default, Nov 22 2022, 08:49:06)
[Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pinocchio as pin
>>> pin.GeometryObject
<class 'pinocchio.deprecated.GeometryObject'>
```

Since `GeometryObject` belongs to `pinocchio.deprecated` I thought it was deprecated. 

The second option looks more appealing to me.

**Reply by jorisv** [reply]
@lwygzh Apparently the method `GeometyObject.fcl()` is deprecated.
To add the deprecated decorator to this method, we redefine `GeometryObject` in the [deprecated module](https://github.com/stack-of-tasks/pinocchio/blob/4512214cf08e765a28542c75bf6378e934622a38/bindings/python/pinocchio/deprecated.py#L89-L93).
This doesn't mean that `GeometryObject` is deprecated (only the fcl method is).

I will discuss the GeometryObject modification with my colleagues and open an issue if they accept it.

**Reply by lwygzh** [reply]
Thanks!
